graph/src/data/subgraph/schema.rs

```diff
@@ -163,18 +164,52 @@ pub struct SubgraphManifestEntity {
     pub repository: Option<String>,
     pub features: Vec<String>,
     pub schema: String,
+    pub raw_yaml: Option<String>,
 }
 
-impl<'a, C: Blockchain> From<&'a super::SubgraphManifest<C>> for SubgraphManifestEntity {
-    fn from(manifest: &'a super::SubgraphManifest<C>) -> Self {
+impl SubgraphManifestEntity {
+    pub fn new(raw_yaml: String, manifest: &super::SubgraphManifest<impl Blockchain>) -> Self {
         Self {
             spec_version: manifest.spec_version.to_string(),
             description: manifest.description.clone(),
             repository: manifest.repository.clone(),
             features: manifest.features.iter().map(|f| f.to_string()).collect(),
             schema: manifest.schema.document.clone().to_string(),
+            raw_yaml: Some(raw_yaml),
         }
     }
+
+    pub fn template_idx_and_name(&self) -> Result<Vec<(i32, String)>, Error> {
```

Comment:
I understand we're using `i32` to simplify comparing these values with query results, but it would be nice to postpone the conversion and create a type alias for this tuple.